### PR TITLE
Added Tutorial on writing a producer

### PR DESCRIPTION
**Pull request by harrypuuter**
You can build the docs locally using https://crown.readthedocs.io/en/latest/introduction.html#creating-documentation

### CODE REVIEW

File: docs/sphinx_source/contrib.rst [hunk 1]
```diff
@@ -1,7 +1,175 @@
...
+                        const int &index) {
+        auto build_vector = [index](ROOT::RVec<float> pts, ROOT::RVec<float> etas,
+                                    ROOT::RVec<float> phis,
+                                    ROOT::RVec<float> masses) {
+            // lambda code here
+        };
+        return df.Define(outputname, build_vector, {pts, etas, phis, masses});
+    };
+
+We define the lambda function with the type ``auto`` and pass it all our input columns. Since this is now the event-by-event part, we have to specify the correct type of the columns. In our case, since we take the quantities directly from the NanoAOD file, they all have the type ``ROOT::RVec<float>``. For the actual implementation, we can simply define a new object with type ``ROOT::Math::PtEtaPhiMVector`` and the ``ROOT`` backend will do the rest. Since we only want to get the lorentzvector for a single electron, we can use ``pts.at(index, -10.)`` to select the correct ``float`` value. If this value cannot be found, e.g. when there is no electron within the event, the default value ``-10`` is used. After adding this implemention, our function is complete and should look like this:
```
Comment (ralfschmieder) [inline docs/sphinx_source/contrib.rst:64]: Lorentz vector

File: docs/sphinx_source/contrib.rst [hunk 2]
```diff
@@ -1,7 +1,175 @@
...
+
+    configuration.add_config_parameters(
+        ["et"],
+        {
+            "electron_index_to_use": 0,
+        },
+    )
+
+
+The last keyword ``scopes`` is used to specify the scopes in which the producer is available. This makes sence in order to prevent errors, where the producer is used in a scope that is not specified, e.g. in a final state without any electrons, we would not need to run this producer. Note that the output of this producer is of type ``ROOT::Math::PtEtaPhiMVector`` so it always makes sence to represent that in the name of the quantity in some way for easier understanding.
```
Comment (ralfschmieder) [inline docs/sphinx_source/contrib.rst:133]: sense

**Review by ralfschmieder — APPROVED** [review]

File: docs/sphinx_source/contrib.rst (hunk 1 above)
Comment (harrypuuter) [reply docs/sphinx_source/contrib.rst]: fixed

File: docs/sphinx_source/contrib.rst (hunk 2 above)
Comment (harrypuuter) [reply docs/sphinx_source/contrib.rst]: fixed

**Pull request merged**